src/api/extension/index.js

```diff
@@ -676,32 +676,16 @@ export const bytesAddressToBinary = (bytes) =>
   bytes.reduce((str, byte) => str + byte.toString(2).padStart(8, '0'), '');
 
 export const isValidAddress = async (address) => {
-  await Loader.load();
   const network = await getNetwork();
-  try {
-    const addr = Loader.Cardano.Address.from_bech32(address);
-    if (
-      (addr.network_id() === 1 && network.id === NETWORK_ID.mainnet) ||
-      (addr.network_id() === 0 &&
-        (network.id === NETWORK_ID.testnet ||
-          network.id === NETWORK_ID.preview ||
-          network.id === NETWORK_ID.preprod))
-    )
-      return addr.to_raw_bytes();
-    return false;
-  } catch (e) {}
-  try {
-    const addr = Loader.Cardano.ByronAddress.from_base58(address);
-    if (
-      (addr.network_id() === 1 && network.id === NETWORK_ID.mainnet) ||
-      (addr.network_id() === 0 &&
-        (network.id === NETWORK_ID.testnet ||
-          network.id === NETWORK_ID.preview ||
-          network.id === NETWORK_ID.preprod))
-    )
-      return addr.to_address().to_raw_bytes();
-    return false;
-  } catch (e) {}
+  const addr = Cardano.Address.fromString(address)
```

Comment:
it returns `null` when provided an invalid address - `addr.getNetworkId()` will throw `Cannot read properties of null...`
```ts
if (!addr) return false;
```

Reply:
https://github.com/input-output-hk/nami/pull/948